tests/std/tests/P0896R4_ranges_alg_unique/test.cpp

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <algorithm>
+#include <cassert>
+#include <concepts>
+#include <ranges>
+#include <span>
+#include <utility>
+
+#include <range_algorithm_support.hpp>
+using namespace std;
+using P = pair<int, int>;
+
+// Validate dangling story
+STATIC_ASSERT(same_as<decltype(ranges::unique(borrowed<false>{})), ranges::dangling>);
+STATIC_ASSERT(same_as<decltype(ranges::unique(borrowed<true>{})), ranges::subrange<int*>>);
+
+struct instantiator {
+    static constexpr P expected[4] = {{0, 99}, {1, 47}, {3, 99}, {4, 47}};
+
+    template <ranges::forward_range ReadWrite>
+    static constexpr void call() {
+        using ranges::unique, ranges::subrange, ranges::iterator_t;
+
+        size_t comparisonCounter = 0;
+        constexpr auto countedEq = [&comparisonCounter](const int a, const int b) {
+            ++comparisonCounter;
+            return a == b;
+        };
+
+        { // Validate iterator + sentinel overload
+            P input[6] = {{0, 99}, {1, 47}, {2, 47}, {3, 99}, {4, 47}, {5, 47}};
+            ReadWrite wrapped_input{input};
+
+            auto result = unique(wrapped_input.begin(), wrapped_input.end(), countedEq, get_second);
+            STATIC_ASSERT(same_as<decltype(result), subrange<iterator_t<ReadWrite>>>);
+            assert(result.begin() == next(wrapped_input.begin(), 4));
+            assert(result.end() == wrapped_input.end());
+            assert(ranges::equal(expected, span{input}.first<4>()));

```

Comment:
Why bother with the `span` schtick instead of simply passing `result`?

Reply:
Because I forgot taht I get a subrange back
```suggestion
            assert(ranges::equal(expected, result));
```

Reply:
Ah damn it. I need to think before pressing that sweet little apply suggestion button. 

The problem is that result equals the range `[_End, _Last)` and not the range `[_First, _End)`

Reply:
Oh, yes, because the returned range is the range of removed elements.